pallets/confidential-asset/src/lib.rs

```diff
@@ -151,6 +132,56 @@ decl_module! {
             Ok(())
         }
 
+        /// Initializes a new confidential security token.
+        /// Makes the initiating account the owner of the security token
+        /// & the balance of the owner is set to total supply.
+        ///
+        /// # Arguments
+        /// * `origin` - contains the secondary key of the caller (i.e who signed the transaction to execute this function).
+        /// * `name` - the name of the token.
+        /// * `ticker` - the ticker symbol of the token.
+        /// * `total_supply` - the total supply of the token.
+        /// * `divisible` - a boolean to identify the divisibility status of the token.
+        /// * `asset_type` - the asset type.
+        /// * `identifiers` - a vector of asset identifiers.
+        /// * `funding_round` - name of the funding round.
+        ///
+        /// # Errors
+        /// - `TickerAlreadyRegistered` if the ticker was already registered, e.g., by `origin`.
+        /// - `TickerRegistrationExpired` if the ticker's registration has expired.
+        /// - `InvalidTotalSupply` if not `divisible` but `total_supply` is not a multiply of unit.
+        /// - `TotalSupplyAboveLimit` if `total_supply` exceeds the limit.
+        /// - `BadOrigin` if not signed.
+        ///
+        /// # Weight
+        /// `3_000_000_000 + 20_000 * identifiers.len()`
+        #[weight = 3_000_000_000 + 20_000 * u64::try_from(identifiers.len()).unwrap_or_default()]
+        pub fn create_confidential_asset(
+            origin,
+            name: AssetName,
+            ticker: Ticker,
+            total_supply: T::Balance,
+            divisible: bool,
+            asset_type: AssetType,
+            identifiers: Vec<(IdentifierType, AssetIdentifier)>,
+            funding_round: Option<FundingRoundName>,
+        ) -> DispatchResult {
+            let primary_owner = ensure_signed(origin)?;
+            let primary_owner_did = Context::current_identity_or::<Identity<T>>(&primary_owner)?;
+
+            T::Asset::unsafe_create_asset(primary_owner_did, name, ticker, total_supply, divisible, asset_type, identifiers, funding_round, true)?;
+
+            // Append the ticker to the list of confidential tickers.
+            let mut token_list = Self::confidential_tickers();
+            token_list.push(AssetId{id: ticker.as_bytes().clone()});
+            <ConfidentialTickers>::put(token_list);
+
+            // TODO(CRYP-160) : mint `total_supply` assets to the primary asset issuer here.
+
+            Self::deposit_event(Event::ConfidentialAssetCreated(primary_owner_did, ticker));
```

Comment:
this event should be similar to the `AssetCreated` event and `AssetCreated` should not be emitted for the non-confidential asset.